src/test/groovy/pl/tomaszdziurko/jvm_bloggers/view/login/attack/stream/BruteForceAttackEventStreamSpec.groovy

```diff
@@ -0,0 +1,72 @@
+package pl.tomaszdziurko.jvm_bloggers.view.login.attack.stream
+
+import org.apache.commons.lang3.tuple.Pair
+import pl.tomaszdziurko.jvm_bloggers.view.login.attack.BruteForceAttackEvent
+import rx.Observable
+import rx.Observer
+import rx.observers.TestSubscriber
+import rx.subjects.Subject
+import spock.lang.Specification
+
+import java.lang.Void as Should
+
+/**
+ * @author Adam Dec
+ * @since 0.7.0
+ */
+class BruteForceAttackEventStreamSpec extends Specification {
+
+    Subject<BruteForceAttackEvent, BruteForceAttackEvent> subject;
+    Observable<Pair<String, String>> observable;
+    BruteForceAttackEventStream stream;
+
+    def setup() {
+        subject = Mock(Subject)
+        observable = Observable.empty()
+        observable.doOnSubscribe() >> {}
+        stream = new BruteForceAttackEventStream(subject, observable)
+    }
+
+    Should "Not publish BruteForceAttackEvent to observers"() {
+        given:
+        subject.hasObservers() >> false
```

Comment:
Please fix indentations so section under each label (given, when, then) is one level deeper.